scvi/external/sysvi/_priors.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import abc
+from abc import abstractmethod
+
+import torch
+from torch.distributions import Normal, kl_divergence
+
+
+class Prior(torch.nn.Module, abc.ABC):
+    @abstractmethod
+    def kl(self, m_q, v_q, z):
+        pass
+
+
+class StandardPrior(Prior):
+    def kl(self, m_q, v_q, z=None):
+        # 1 x N
+        return kl_divergence(
+            Normal(m_q, v_q.sqrt()), Normal(torch.zeros_like(m_q), torch.ones_like(v_q))
+        ).sum(dim=1)
+
+
+class VampPrior(Prior):
+    """VampPrior adapted from https://github.com/jmtomczak/intro_dgm/main/vaes/vae_priors_example.ipynb
```

Comment:
I'm getting an error that this link does not exist

Reply:
updated